.github/workflows/test.yml

```
@@ -1,5 +1,11 @@
 name: test
-on: [push]
+on:
+  pull_request:
+    branches: [main]
```

```suggestion
on:
  push:
    branches:
      - main
  pull_request:
    types: [opened, synchronize]
```

I'd usually do this. Run it on push on the default branch for CI, and limit types of pull request events. I'd still run CI even if we send PRs against other branches.

---

By default, a workflow only runs when a `pull_request` event's activity type is `opened`, `synchronize`, or `reopened`, so I edited your suggestion to remove `types` in favor of the defaults.